qiskit_experiments/library/calibration/frequency_cal.py

```diff
@@ -0,0 +1,112 @@
+
+    # section: see_also
+        qiskit_experiments.library.characterization.ramsey_xy.RamseyXY
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        calibrations: BackendCalibrations,
+        backend: Optional[Backend] = None,
+        delays: Optional[List] = None,
+        unit: str = "s",
+        osc_freq: float = 2e6,
+        auto_update: bool = True,
+    ):
+        """
+        Args:
+            qubit: The qubit on which to run the frequency calibration.
+            calibrations: The calibrations instance with the schedules.
+            backend: Optional, the backend to run the experiment on.
+            delays: The list of delays that will be scanned in the experiment.
+            unit: The unit of the delays. Accepted values are dt, i.e. the
+                duration of a single sample on the backend, seconds, and sub-units,
+                e.g. ms, us, ns.
+            osc_freq: A frequency shift in Hz that will be applied by means of
+                a virtual Z rotation to increase the frequency of the measured oscillation.
+            auto_update: If set to True, which is the default, then the experiment will
+                automatically update the frequency in the calibrations.
+        """
+        super().__init__(
+            calibrations,
+            qubit,
+            backend=backend,
+            delays=delays,
+            unit=unit,
+            osc_freq=osc_freq,
+            cal_parameter_name="qubit_lo_freq",
+            updater=Frequency,
+            auto_update=auto_update,
+        )
+
+        # Instruction schedule map to bring in the calibrations for the sx gate.
+        self.set_transpile_options(inst_map=calibrations.default_inst_map)
+
+    def _add_cal_metadata(self, circuits: List[QuantumCircuit]):
+        """Add the oscillation frequency of the experiment to the metadata."""
+
+        param_val = self._cals.get_parameter_value(
+            self._param_name,
+            self.physical_qubits,
+            group=self.experiment_options.group,
+        )
+
+        for circuit in circuits:
+            circuit.metadata["cal_param_value"] = param_val
+            circuit.metadata["cal_group"] = self.experiment_options.group
+            circuit.metadata["osc_freq"] = self.experiment_options.osc_freq
+
+    def update_calibrations(self, experiment_data: ExperimentData):
+        """Update the frequency using the reported frequency less the imparted oscillation."""
+
+        data = experiment_data.data()
+
+        # No data -> no update
+        if len(data) > 0:
+            result_index = self.experiment_options.result_index
+            osc_freq = data[0]["metadata"]["osc_freq"]
+            group = data[0]["metadata"]["cal_group"]
+            old_freq = data[0]["metadata"]["cal_param_value"]
+
+            fit_freq = BaseUpdater.get_value(experiment_data, "freq", result_index)
+            new_freq = old_freq + fit_freq - osc_freq
```

Comment:
Can you also add this to `self.analysis_options.extra` as dictionary? https://github.com/Qiskit/qiskit-experiments/pull/488
I think `osc_freq` is something worth seeing in the database.

Or you can directly add new result entry with the value `fit_freq - osc_freq`.

Reply:
Ignore this. The extra option of curve analysis will be removed.

Reply:
I have been trying to use the experiments service more, and I agree generally that it would be good to get more metadata displayed there.